delta-lake/src/main/java/org/apache/iceberg/delta/BaseMigrateDeltaLakeTableAction.java

```diff
@@ -0,0 +1,275 @@
+        // Delete_Files case
+        DeleteFiles deleteFiles = table.newDelete();
+        filesToRemove.forEach(deleteFiles::deleteFile);
+        deleteFiles.commit();
+      }
+    }
+  }
+
+  private DataFile buildDataFileFromAction(Action action, Table table, PartitionSpec spec) {
+    String path;
+    Optional<Long> nullableSize;
+    Map<String, String> partitionValues;
+    long size;
+
+    if (action instanceof AddFile) {
+      AddFile addFile = (AddFile) action;
+      path = addFile.getPath();
+      nullableSize = Optional.of(addFile.getSize());
+      partitionValues = addFile.getPartitionValues();
+    } else if (action instanceof RemoveFile) {
+      RemoveFile removeFile = (RemoveFile) action;
+      path = removeFile.getPath();
+      nullableSize = removeFile.getSize();
+      partitionValues = removeFile.getPartitionValues();
+    } else {
+      throw new IllegalStateException(
+          String.format(
+              "Unexpected action type for Delta Lake: %s", action.getClass().getSimpleName()));
+    }
+
+    if (partitionValues == null) {
+      // For unpartitioned table, the partitionValues should be an empty map rather than null
+      throw new ValidationException("File %s does not specify a partitionValues", path);
+    }
+
+    String fullFilePath = deltaLog.getPath().toString() + File.separator + path;
+    FileFormat format = determineFileFormatFromPath(fullFilePath);
+
+    Metrics metrics =
+        FileMetricsReader.getMetricsForFile(table, fullFilePath, format, this.hadoopConfiguration);
+    String partition =
+        spec.fields().stream()
+            .map(PartitionField::name)
+            .map(name -> String.format("%s=%s", name, partitionValues.get(name)))
+            .collect(Collectors.joining("/"));
+
+    size = nullableSize.orElseGet(() -> table.io().newInputFile(path).getLength());
+
+    return DataFiles.builder(spec)
+        .withPath(fullFilePath)
+        .withFormat(format)
+        .withFileSizeInBytes(size)
+        .withMetrics(metrics)
+        .withPartitionPath(partition)
+        .build();
+  }
+
+  private FileFormat determineFileFormatFromPath(String path) {
+    if (path.endsWith(parquetSuffix)) {
+      return FileFormat.PARQUET;
+    } else if (path.endsWith(avroSuffix)) {
+      return FileFormat.AVRO;
+    } else if (path.endsWith(orcSuffix)) {
+      return FileFormat.ORC;
+    } else {
+      throw new ValidationException("The format of the file %s is unsupported", path);
+    }
+  }
+
+  private static Map<String, String> destTableProperties(
+      io.delta.standalone.Snapshot deltaSnapshot,
+      String tableLocation,
+      Map<String, String> additionalProperties) {
+    Map<String, String> properties = Maps.newHashMap();
+
+    properties.putAll(deltaSnapshot.getMetadata().getConfiguration());
+    properties.putAll(
+        ImmutableMap.of(
+            "migration_source", "delta", "table_type", "iceberg", "location", tableLocation));
```

Comment:
do we really need `table_type` and `location`? I think catalog will add those when creating the table if necessary.

Reply:
Thank you for your suggestion. I agree we may not need `table_type` because the `icebergCatalog` implementation will add it to the table properties if needed. For `location`, I chose to keep it because I wanted to let user have track on the original location of the delta lake table, since all `parquet` datafiles will still be there after the migration. I renamed this property as "original_location". Do you think it is ok to have this remained in the table properties?